### PR TITLE
Icds reports ls tabular report

### PR DESCRIPTION
**Pull request by dmydlo**
Hi @calellowitz,
I have developed the Lady Supervisor tabular report.
It requires aggregating awc_locations.
These changes regard this JIRA [ticket](https://dimagi-dev.atlassian.net/browse/ICDS-408).
Need QA: Yes
Environment: n/a
Locally Tested: Yes
Regards, Dawid

### CODE REVIEW

**Comment by dmydlo** [comment]
I have to fix tests that broke after merging with master.

File: custom/icds_reports/tasks.py [hunk 1]
```diff
@@ -693,7 +695,25 @@ def prepare_excel_reports(config, aggregation_level, include_test, beta, locatio
             )
         else:
             cache_key = create_excel_file(excel_data, data_type, file_format)
-    if indicator != AWW_INCENTIVE_REPORT:
+    elif indicator == LS_REPORT_EXPORT:
+        data_type = 'Lady_Supervisor'
+        config['aggregation_level'] = 4  # this report is explicitly not aggregated
```
Comment (Rohit25negi) [inline custom/icds_reports/tasks.py]: Can you make this comment clearer. First glance  looks that the data is not aggregated but the code will do that later. I think something like 'this report is not to be aggregated' or something similar will be good for people who does not know the spec of this report

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
Hey, Left a few comments. Also restarted the failing test case.

File: custom/icds_reports/sqldata/exports/lady_supervisor.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    table_name = 'agg_ls_monthly'
+
+    @property
+    def get_columns_by_loc_level(self):
+        return [
+            DatabaseColumn('State', SimpleColumn('state_name')),
+            DatabaseColumn('District', SimpleColumn('district_name'), slug='district_name'),
+            DatabaseColumn('Block', SimpleColumn('block_name'), slug='block_name'),
+            DatabaseColumn('Sector Name', SimpleColumn('supervisor_name'), slug='supervisor_name'),
+            DatabaseColumn('Name of Lady Supervisor', SimpleColumn('ls_name'), slug='ls_name'),
```
[the remark this answers is not in the document]
Comment (dmydlo) [reply custom/icds_reports/sqldata/exports/lady_supervisor.py]: Ok, sure

File: custom/icds_reports/static/js/directives/download/download.directive.js [hunk 3]
```diff
@@ -462,7 +463,8 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
     };
 
     vm.isVisible = function(level) {
```
[the remark this answers is not in the document]
Comment (dmydlo) [reply custom/icds_reports/static/js/directives/download/download.directive.js]: You are right, I will add it to vm.indicators.

File: custom/icds_reports/migrations/0071_visits_not_vists.py [hunk 4]
```diff
@@ -14,13 +14,4 @@ class Migration(migrations.Migration):
     dependencies = [
         ('icds_reports', '0070_aww_name_in_agg_ccs_view'),
     ]
-    sql_views = [
```
[the remark this answers is not in the document]
Comment (dmydlo) [reply custom/icds_reports/migrations/0071_visits_not_vists.py]: Well, those actions build sql views, yet those views require awc_location_monthly view to be present, but then awc_location_monthly needs awc_location table with ls_name that isn't going to be present until last migration.
This way until ls_name isn't in awc_location, views can't be created.

File: custom/icds_reports/tasks.py (hunk 1 above)
Comment (dmydlo) [reply custom/icds_reports/tasks.py]: I am going to change it to: `this report on all levels shows data (row) per sector`

File: custom/icds_reports/sqldata/exports/lady_supervisor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmydlo) [reply custom/icds_reports/sqldata/exports/lady_supervisor.py]: FYI: Also, this was due to finding out the problem with migrations later in the process.

File: custom/icds_reports/tasks.py (hunk 1 above)
Comment (Rohit25negi) [reply custom/icds_reports/tasks.py]: Sounds good

File: custom/icds_reports/static/js/directives/download/download.directive.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Rohit25negi) [reply custom/icds_reports/static/js/directives/download/download.directive.js]: Thanks

File: custom/icds_reports/migrations/0071_visits_not_vists.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rohit25negi) [reply custom/icds_reports/migrations/0071_visits_not_vists.py]: I think If you would not have put the service delivery monthly view [here](https://github.com/dimagi/commcare-hq/pull/23661/files#diff-a94be1dc13e319f3f20e10c9db99f493R19). And Just add the migration file mannually for this after the other new migration. you would not have to touch the previous migrations.

[the remark this answers is not in the document]
Comment (dmydlo) [reply custom/icds_reports/migrations/0071_visits_not_vists.py]: If anywhere prior to the last migration there would be either creating `awc_location_monthly` view or any dependent on it, it wouldn't work.

[the remark this answers is not in the document]
Comment (calellowitz) [reply custom/icds_reports/migrations/0071_visits_not_vists.py]: @Rohit25negi 
Dawid is correct, changes to these views break the old migrations. It is a crappy situation, but for now we have to continue to do this.

File: custom/icds_reports/sqldata/exports/lady_supervisor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (calellowitz) [reply custom/icds_reports/sqldata/exports/lady_supervisor.py]: Why is this using sqlagg? All of this should be doable with the django orm and will be much easier to read and maintain.

**Review by calellowitz — CHANGES_REQUESTED** [review]
Generally fine. Added a few comments

**Comment by dmydlo** [comment]
Hi, @calellowitz,
I have looked into what would it take to not use sqlagg and I have to disagree that it would be easier to maintain.
The reason is, that we are requiring usage of sqlagg in the mixins from `corehq.apps.reports.sqlreport` and `custom.icds_reports.utils.mixins`. That would mean that rather than having 8 columns using sqlagg, we would have to redo all the filtering, ordering, as well as the creation of headers, fallbacks to `Data not Entered` and location data searching which would end up as hardcoded 80% copy of `ExportableMixin`. I don't think that this would be a preferable option to come up with a different approach for files in sqldata.
Regards, Dawid

**Review by calellowitz — APPROVED** [review]

**Comment by calellowitz** [comment]
@dmydlo 
I see what you are saying about exports specifically. Fine to continue this way for now (although long term, i would love to migrate all this away from sqlagg)

**Review by Rohit25negi — APPROVED** [review]

**Pull request merged**